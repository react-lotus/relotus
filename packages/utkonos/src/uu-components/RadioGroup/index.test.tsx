/* eslint-disable import/no-extraneous-dependencies */
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { createRef } from 'react';
import { RadioGroup } from '../index';

describe('Uikit/RadioGroup', () => {
  describe('snapshot', () => {
    it('default', () => {
      const component = (
        <RadioGroup
          options={[
            { value: 'value1', label: 'label1' },
            { value: 'value2', label: 'label2' },
          ]}
        />
      );

      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('disabled', () => {
      const component = (
        <RadioGroup
          disabled
          options={[
            { value: 'value1', label: 'label1' },
            { value: 'value2', label: 'label2' },
          ]}
        />
      );

      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('only one item disabled', () => {
      const component = (
        <RadioGroup
          options={[
            { value: 'value1', label: 'label1' },
            { value: 'value2', label: 'label2', disabled: true },
          ]}
        />
      );

      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('default value', () => {
      const component = (
        <RadioGroup
          value="value1"
          options={[
            { value: 'value1', label: 'label1' },
            { value: 'value2', label: 'label2' },
          ]}
        />
      );

      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('component', () => {
    it('should properly pass ref to the RadioGroup element', () => {
      const ref = createRef<HTMLDivElement>();
      const { container } = render(
        <RadioGroup
          options={[
            { value: 'value1', label: 'label1' },
            { value: 'value2', label: 'label2' },
          ]}
          ref={ref}
        />,
      );
      ref.current?.focus();
      const divWithRef = container.getElementsByClassName('uu-radioGroup');
      expect(divWithRef[0]).toHaveFocus();
    });
  });
});
