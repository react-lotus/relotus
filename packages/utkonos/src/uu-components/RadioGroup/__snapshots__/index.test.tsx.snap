// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uikit/RadioGroup snapshot default 1`] = `
<div
  className="uu-radioGroup uu-radioGroup_direction_row"
  tabIndex={-1}
>
  <label
    aria-label="label1"
    className="uu-radio"
    htmlFor="nanoid"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <input
      checked={false}
      className="uu-radioInput"
      id="nanoid"
      name="nanoid"
      type="radio"
      value="value1"
    />
    <span
      className="uu-radioMark"
    >
      <span
        className="uu-radioMarkPoint"
      />
    </span>
    <span
      className="uu-radioLabel"
    >
      label1
    </span>
  </label>
  <label
    aria-label="label2"
    className="uu-radio"
    htmlFor="nanoid"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <input
      checked={false}
      className="uu-radioInput"
      id="nanoid"
      name="nanoid"
      type="radio"
      value="value2"
    />
    <span
      className="uu-radioMark"
    >
      <span
        className="uu-radioMarkPoint"
      />
    </span>
    <span
      className="uu-radioLabel"
    >
      label2
    </span>
  </label>
</div>
`;

exports[`Uikit/RadioGroup snapshot default value 1`] = `
<div
  className="uu-radioGroup uu-radioGroup_direction_row"
  tabIndex={-1}
>
  <label
    aria-label="label1"
    className="uu-radio"
    htmlFor="nanoid"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <input
      checked={true}
      className="uu-radioInput"
      id="nanoid"
      name="nanoid"
      type="radio"
      value="value1"
    />
    <span
      className="uu-radioMark"
    >
      <span
        className="uu-radioMarkPoint uu-radioMarkPoint_checked"
      />
    </span>
    <span
      className="uu-radioLabel"
    >
      label1
    </span>
  </label>
  <label
    aria-label="label2"
    className="uu-radio"
    htmlFor="nanoid"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <input
      checked={false}
      className="uu-radioInput"
      id="nanoid"
      name="nanoid"
      type="radio"
      value="value2"
    />
    <span
      className="uu-radioMark"
    >
      <span
        className="uu-radioMarkPoint"
      />
    </span>
    <span
      className="uu-radioLabel"
    >
      label2
    </span>
  </label>
</div>
`;

exports[`Uikit/RadioGroup snapshot disabled 1`] = `
<div
  className="uu-radioGroup uu-radioGroup_direction_row"
  tabIndex={-1}
>
  <label
    aria-label="label1"
    className="uu-radio uu-radio_disabled"
    htmlFor="nanoid"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <input
      checked={false}
      className="uu-radioInput"
      disabled={true}
      id="nanoid"
      name="nanoid"
      type="radio"
      value="value1"
    />
    <span
      className="uu-radioMark"
    >
      <span
        className="uu-radioMarkPoint uu-radioMarkPoint_disabled"
      />
    </span>
    <span
      className="uu-radioLabel uu-radioLabel_disabled"
    >
      label1
    </span>
  </label>
  <label
    aria-label="label2"
    className="uu-radio uu-radio_disabled"
    htmlFor="nanoid"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <input
      checked={false}
      className="uu-radioInput"
      disabled={true}
      id="nanoid"
      name="nanoid"
      type="radio"
      value="value2"
    />
    <span
      className="uu-radioMark"
    >
      <span
        className="uu-radioMarkPoint uu-radioMarkPoint_disabled"
      />
    </span>
    <span
      className="uu-radioLabel uu-radioLabel_disabled"
    >
      label2
    </span>
  </label>
</div>
`;

exports[`Uikit/RadioGroup snapshot only one item disabled 1`] = `
<div
  className="uu-radioGroup uu-radioGroup_direction_row"
  tabIndex={-1}
>
  <label
    aria-label="label1"
    className="uu-radio"
    htmlFor="nanoid"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <input
      checked={false}
      className="uu-radioInput"
      id="nanoid"
      name="nanoid"
      type="radio"
      value="value1"
    />
    <span
      className="uu-radioMark"
    >
      <span
        className="uu-radioMarkPoint"
      />
    </span>
    <span
      className="uu-radioLabel"
    >
      label1
    </span>
  </label>
  <label
    aria-label="label2"
    className="uu-radio uu-radio_disabled"
    htmlFor="nanoid"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <input
      checked={false}
      className="uu-radioInput"
      disabled={true}
      id="nanoid"
      name="nanoid"
      type="radio"
      value="value2"
    />
    <span
      className="uu-radioMark"
    >
      <span
        className="uu-radioMarkPoint uu-radioMarkPoint_disabled"
      />
    </span>
    <span
      className="uu-radioLabel uu-radioLabel_disabled"
    >
      label2
    </span>
  </label>
</div>
`;
